File: src/main/java/br/com/zumbolovsky/fateapp/web/LoginController.java
package br.com.zumbolovsky.fateapp.web;

import org.springframework.http.HttpHeaders;
import org.springframework.http.ResponseEntity;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import static br.com.zumbolovsky.fateapp.config.security.JwtToken.generateAccessToken;

@RestController
class LoginController {

    private final AuthenticationManager authenticationManager;

    public LoginController(AuthenticationManager authenticationManager) {
        this.authenticationManager = authenticationManager;
    }

    @PostMapping("/login")
    public ResponseEntity<Void> login(@RequestBody @Validated AuthRequest request) {
        final Authentication authentication = authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(request.getUser(), request.getPassword()));
        return ResponseEntity.ok()
                .header(HttpHeaders.AUTHORIZATION, generateAccessToken(authentication.getName()))
                .build();
    }
}

File: src/main/kotlin/br/com/zumbolovsky/fateapp/web/LoginController.kt
package br.com.zumbolovsky.fateapp.web

import br.com.zumbolovsky.fateapp.config.security.JwtToken
import br.com.zumbolovsky.fateapp.domain.postgres.UserInfo
import org.springframework.http.HttpHeaders
import org.springframework.http.ResponseEntity
import org.springframework.security.authentication.AuthenticationManager
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class LoginController(private val authenticationManager: AuthenticationManager) {

    @PostMapping("/login")
    fun login(@RequestBody @Validated request: AuthRequest): ResponseEntity<Nothing> =
        authenticationManager.authenticate(UsernamePasswordAuthenticationToken(request.user, request.password))
            .let {
                ResponseEntity.ok()
                    .header(
                        HttpHeaders.AUTHORIZATION,
                        JwtToken.generateAccessToken(
                            UserInfo(
                                user = it.name,
                                password = it.credentials as String)))
                    .build()
            }
}